src/librustc_typeck/check/mod.rs

```diff
@@ -2481,16 +2481,16 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                     if arg_count == 1 {" was"} else {"s were"}),
                 error_code);
 
-            err.span_label(sp, &format!("expected {}{} parameter{}",
-                                        if variadic {"at least "} else {""},
-                                        expected_count,
-                                        if expected_count == 1 {""} else {"s"}));
-
             let input_types = fn_inputs.iter().map(|i| format!("{:?}", i)).collect::<Vec<String>>();
             if input_types.len() > 0 {
-                err.note(&format!("the following parameter type{} expected: {}",
-                        if expected_count == 1 {" was"} else {"s were"},
-                        input_types.join(", ")));
+                err.span_label(sp, &format!("the following parameter type{} expected: {}",
```

Comment:
If we're going to make this a label, can we make it smaller?  Maybe something like it simply:

"parameter type{} expected: {}"

Makes we wish we had a span_label for the definition site, so they could see where the requirements were coming from (if we don't have that already)